CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,81 @@
+# PureScript Community Code of Conduct
+
+The PureScript community follows some simple ground rules to protect safe, inclusive spaces to discuss the PureScript programming language.
+
+All community members are expected to act and interact in ways that contribute to an open, welcoming, diverse, inclusive, and healthy community.
+
+## What's our purpose?
+
+The PureScript community channels are places to discuss the PureScript programming language. 
+
+* We talk about the language, its related libraries and infrastructure, and our experiences using it.
+* We help PureScript programmers learn the language, and we ask for help becoming stronger programmers ourselves. 
+* We discuss general functional programming concepts and other parts of programming, type theory, and mathematics that are relevant to Purescript.
+* We organize open-source work, meetups, hackathons, and other community events.
+* We discuss what we can do to grow an inclusive, friendly, respectful community.
+
+## What is okay?
+
+Examples of behavior that contributes to a positive environment for our community include:
+
+* Demonstrating empathy and kindness toward other people.
+* Being respectful of differing opinions, viewpoints, and experiences.
+* Giving and gracefully accepting constructive feedback.
+* Accepting responsibility and apologizing to those affected by our mistakes, and learning from the experience.
+
+As participants in the PureScript community, we pledge to make participation in our community a harassment-free experience for everyone -- regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, caste, color, religion, or sexual identity and orientation.
```

Comment:
Split "gender identity and expression" into "gender identity, gender expression". Similarly "religion, or sexual identity and orientation" becomes "religion, sexual identity or sexual orientation."

Reply:
I think those are intentionally grouped together.

Reply:
I think that they probably were originally grouped together intentionally. But I believe splitting them is worthwhile for two reasons.
1. By grouping "gender identity and expression" and "sexual identity and orientation" we are saying that identity and expression in those instances are tied together in a way that, for example, race and color are not. That may be true -- it's a long way outside my field of expertise. But what we are doing with this sentence is highlighting examples and I think it would be better to be silent on the relationships between any of these attributes. We might also want to sort the list in alphabetical order to remove any hint of hierarchy between them.

2. No reasonable person would make the argument that "I was only harassing this person based on their sexual orientation, not their sexual identity. Therefore I was not harassing them based on their "sexual identity and sexual orientation" because 'false and true equals false'". But if there ever was a person who would make that argument it would be a troll with a background in computer science.

Reply:
I think this list should be removed altogether . You're never gonna get it right and it's logically the same statement if you remove it: "we pledge to make participation in our community a harassment-free experience for everyone."

Reply:
Strongly disagree. It may be the case that it’ll never be perfect but we can always amend it in the future if we need to. To me, if a CoC doesn’t actually list the protected characteristics like this, it’s a good signal that the community doesn’t recognise one or more of them as being worthy of protection, and this would make me _less_ likely to want to participate.

Reply:
@hdgarrood but doesn't saying "harassment-free" mean no harassment of any kind? Sans the list, you think people will read this and think to themselves "hm, perhaps I am the unstated exception to this rule"?

Reply:
It's that different people have different definitions of what constitutes harassment. A list like this signals that the moderators are (or at least claim to be) aware of the specific concerns of the mentioned communities, and intend to take them into account when determining whether behavior merits moderator action. Sans the list, people will read this and wonder whether ‘harassment-free’ means that the specific, possibly controversial form of harassment that they've experienced or seen others experience is going to get a pass because it's not an overt threat.

Reply:
Okay, this is a very good point. I still don't like the list as is, because to me it logically disguises it's true purpose: to more clearly define what the community considers harassment. Personally, I would prefer it be reworded/reorganized to be inline with this purpose.

Reply:
I've tried to reword things so that this sentiment (define some examples of what the community considers harassment) is more obvious.